build_tools/benchmarks/post_benchmark_comment_test.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from unittest import mock
+import http.client
+import requests
+import unittest
+
+import post_benchmark_comment
+
+
+class PostBenchmarkCommentTest(unittest.TestCase):
+
+  def test_post_to_gist(self):
+    gist_url = "https://example.com/123455/1234.md"
+    response = mock.create_autospec(requests.Response)
+    response.status_code = http.client.CREATED
+    response.json.return_value = {"html_url": gist_url, "truncated": False}
+    requester = mock.create_autospec(post_benchmark_comment.APIRequester)
+    requester.post.return_value = response
+
+    url = post_benchmark_comment.post_to_gist(requester=requester,
+                                              filename="1234.md",
+                                              content="xyz")
+
+    self.assertEqual(url, gist_url)
+    requester.post.assert_called_once_with(
+        endpoint=post_benchmark_comment.GITHUB_GIST_API,
+        payload={
+            "public": True,
+            "files": {
+                "1234.md": {
+                    "content": "xyz"
+                }
+            }
+        })
+
+  def test_post_to_gist_truncated(self):
+    gist_url = "example.com/123455/1234.md"
+    response = mock.create_autospec(requests.Response)
+    response.status_code = http.client.CREATED
+    response.json.return_value = {"html_url": gist_url, "truncated": True}
+    requester = mock.create_autospec(post_benchmark_comment.APIRequester)
+    requester.post.return_value = response
+
+    self.assertRaises(
+        RuntimeError, lambda: post_benchmark_comment.post_to_gist(
+            requester=requester, filename="1234.md", content="xyz"))
```

Comment:
Nit: IMO context managers are much more readable than lambdas, and they allow you to assert things about the error afterwards as well.

```suggestion
    with self.assertRaises(RuntimeError) as e:
        post_benchmark_comment.post_to_gist(
            requester=requester, filename="1234.md", content="xyz"))
```

Reply:
Done.